src/Adapter/AbstractObjectModelRepository.php

```diff
@@ -127,6 +187,32 @@ protected function addObjectModel(ObjectModel $objectModel, string $exceptionCla
         }
     }
 
+    /**
+     * @param ObjectModel $objectModel
+     * @param string $exceptionClass
+     * @param int $errorCode
+     *
+     * @return int
+     */
+    protected function addObjectModelToShop(ObjectModel $objectModel, int $shopId, string $exceptionClass, int $errorCode = 0): int
+    {
+        // Store object's list of shop ID to reset it appropriately after the update
+        $savedShopIds = $objectModel->id_shop_list;
+        $objectModel->id_shop_list = [$shopId];
```

Comment:
Comment this better (and in the ObjectModel as well)

Reply:
Be careful this might change the `id_shop_default` check that we don't mess with it